packages/components-library/tests/unit/filters/StringFilter.spec.ts

```diff
@@ -0,0 +1,58 @@
+import { mount, createLocalVue } from '@vue/test-utils'
+import StringFilter from '@/components/filters/StringFilter.vue'
+import { BootstrapVue } from 'bootstrap-vue'
+
+const localVue = createLocalVue()
+localVue.use(BootstrapVue)
+
+describe('StringFilter.vue', () => {
+  let wrapper: any
+  beforeEach(() => {
+    wrapper = mount(StringFilter, {
+      localVue,
+      propsData: {
+        name: 'name',
+        label: 'label',
+        value: 'value'
+      }
+    })
+  })
+
+  it('matches the snapshot', () => {
+    expect(wrapper.element).toMatchSnapshot()
+  })
+
+  it('sets value property on inner input', () => {
+    const inputElement = wrapper.find('input').element as HTMLInputElement
+    expect(inputElement.value).toBe('value')
+  })
+
+  it('emits undefined when clear button is clicked', () => {
+    wrapper.find('button').trigger('click')
+    // @ts-ignore
+    expect(wrapper.emitted().input[0]).toEqual([undefined])
+  })
+
+  it('should emit an input event when the input value changes', async () => {
+    const input = wrapper.find('input')
+    // @ts-ignore
+    input.element.value = 'change me'
+    await input.trigger('input')
+    // @ts-ignore
+    expect(wrapper.emitted().input[0]).toEqual(['change me'])
+  })
+
+  describe('not passing a model prop', () => {
+    const wrapper = mount(StringFilter, {
+      localVue,
+      propsData: {
+        name: 'name',
+        label: 'label'
+      }
+    })
```

Comment:
Hier een getWrapper van maken, net zoals CheckboxFilter.spec.ts?

Reply:
current solution seems more clear / simpel , agree that is complex setup is need a helper function is nice, in general unit test should try to avoid complex setups imo , if complex setups are needed this to me signals that the unit under test might not we very modulair.